### ISSUE TITLE
Replicate GKX Baselines

### ISSUE THREAD

**Issue by Racoo203** [Racoo203/conditional_asset_pricing#2]
To benchmark our future unsupervised learning models effectively, we need to replicate the core supervised models used in Gu, Kelly, and Xiu (2020). Currently, we only support OLS and ElasticNet. We need to expand the ModelFactory to support non-linear and latent-factor baselines.

**Objectives**:
- Enable Principal Component Regression (PCR) to serve as the linear baseline for latent structure.
- Enable Random Forest and XGBoost to serve as the non-linear performance barrier.
- Enable Neural Networks as a more complex, non-linear models.
- Ensure these models are compatible with the existing ModelConfig and Optuna tuning pipeline.

**Technical Tasks**:
- [ ] A PCR, RandomForest, XGBoost, and MLP factory methods to src/models/config.py.
- [ ] Update src/models/trainer.py's ModelFactory to instantiate these models.
- [ ] Implement sklearn.pipeline.Pipeline logic for PCR (StandardScaler -> PCA -> LinearRegression).
- [ ] Define Optuna search spaces for XGBoost (learning rate, depth) and PCR (n_components). 
- [ ] Add new dependencies to environment.yml.
- [ ] Preprocessing changes may be necessary.

**Acceptance Criteria**:

- Can successfully train and evaluate a PCR model via baselines.py.
- Can successfully tune an XGBoost model using Optuna.
- Results for these models appear in the final experiment log and are similar to what's reported from GKX.
